src/taco-dependency-installer/test/androidSdkInstaller.ts

```diff
@@ -0,0 +1,131 @@
+            // Register mocks. child_process and taco-utils mocks needs to be register before 
+            // AndroidSdkInstaller is required for the mock to work
+            mockery.registerMock("child_process", new nodeFakes.ChildProcessModule().fakeAllExecCallsEndingWithErrors());
+
+            var fakeTelemetryHelper: TacoTestsUtils.TelemetryFakes.Helper = new tacoTestsUtils.TelemetryFakes.Helper();
+            var tacoUtilsWithFakeTelemetry = _.extend({}, tacoUtils, { TelemetryHelper: fakeTelemetryHelper, HasFakes: true });
+            mockery.registerMock("taco-utils", tacoUtilsWithFakeTelemetry);
+
+            // We require the AndroidSdkInstaller file, which will use all the mocked dependencies
+            var androidSdkInstallerClass = require("../installers/androidSdkInstaller");
+
+            // Mocking process messes with node.js, so we do it after requiring all the files
+            var fakeProcess = new nodeFakes.Process().fakeMacOS().fakeDeterministicHrtime().asProcess();
+            // TODO: Try to find a nicer way of doing this. Maybe setting process as protoype of FakeProcess?
+            process = <any> _.extend({}, process, fakeProcess);
+
+            var steps: DependencyInstallerInterfaces.IStepsDeclaration = {
+                download: false,
+                install: false,
+                updateVariables: true, // We only test this step on this test
+                postInstall: false
+            };
+
+            var installerInfo: DependencyInstallerInterfaces.IInstallerData = {
+                installSource: "",
+                sha1: "",
+                bytes: 0,
+                installDestination: "",
+                steps: steps
+            };
+
+            var softwareVersion: string = "";
+            var installTo: string = "";
+
+            var androidSdkInstaller = new androidSdkInstallerClass(installerInfo, softwareVersion, installTo,
+                new FakeLogger(), steps);
+
+            var expectedTelemetry = [
+                {
+                    "initialStep.time": { isPii: false, value: "2000" },
+                    step: { isPii: false, value: "initialStep" }
+                },
+                {
+                    "error.description": { isPii: false, value: "ErrorOnChildProcess on updateVariablesDarwin" },
+                    lastStepExecuted: { isPii: false, value: "updateVariables" },
+                    step: { isPii: false, value: "updateVariables" },
+                    time: { isPii: false, value: "3000" },
+                    "updateVariables.time": { isPii: false, value: "2000" }
+                }
+            ];
+
+            androidSdkInstaller.run()
+                .done(() => done(new Error("Should have gotten a rejection in this test")), () => {
+                    fakeTelemetryHelper.getAllSentEvents().done((allSentEvents: TelemetryEvent[]) => {
+                        allSentEvents.should.eql(expectedTelemetry);
```

Comment:
No. I'm getting this when the assert fails:
  AndroidSdkInstaller telemetry
    updateVariablesDarwin

folder...\applicationinsights\AutoCollection\Exceptions.js:25
                    throw error;
                    ^
AssertionError: expected [
  {
  'initialStep.time': { isPii: false, value: '2000' },
  step: { isPii: false, value: 'initialStep' }
},
  {
  'error.description': {
  isPii: false,
  value: 'ErrorOnChildProcess INVALID MESSAGE on updateVariablesDarwin'
},
  lastStepExecuted: { isPii: false, value: 'updateVariables' },
  step: { isPii: false, value: 'updateVariables' },
  time: { isPii: false, value: '3000' },
  'updateVariables.time': { isPii: false, value: '2000' }
}
] to equal [
  {
  'initialStep.time': { isPii: false, value: '2000' },
  step: { isPii: false, value: 'initialStep' }
},
  {
  'error.description': {
  isPii: false,
  value: 'ErrorOnChildProcess on updateVariablesDarwin'
},
  lastStepExecuted: { isPii: false, value: 'updateVariables' },
  step: { isPii: false, value: 'updateVariables' },
  time: { isPii: false, value: '3000' },
  'updateVariables.time': { isPii: false, value: '2000' }
}
] (at '1' -> 'error.description' -> value, A has 'ErrorOnChildProcess INVALID MESSAGE on updateVariablesDarwin' and B has 'ErrorOnChildProcess on updateVariablesDarwin')
    at Assertion.fail (folder...\should\lib\assertion.js:113:17)
    at Assertion.prop.(anonymous function) [as eql] (folder...\should\lib\assertion.js:39:14)
    at folder...\test\androidSdkInstaller.js:99:42
From previous event:
    at folder...\test\androidSdkInstaller.js:98:56